File: src/MazeSolve/ConsoleUI.cs
public static class ConsoleUI
{
    public static void Init()
    {
        // Allow UTF-8 encoding
        Console.OutputEncoding = System.Text.Encoding.Unicode;
    }

    public static string AskForString(string enquiry)
    {
        Console.Write($"{enquiry}: ");
        return Console.ReadLine();
    }

    public static bool AskYN(string question)
    {
        Console.Write($"{question} (y/n): ");
        var key = Console.ReadKey();
        Console.WriteLine();
        return key.KeyChar == 'y' || key.KeyChar == 'Y';
    }
}

File: src/MazeSolveFS/ConsoleUI.fs
module ConsoleUI

open System

let init () =
  // Set the console encoding to Unicode
  Console.OutputEncoding <- System.Text.Encoding.Unicode

let askForString (enquiry: string) : string =
  printf "%s: " enquiry
  Console.ReadLine()

let askYN (question: string) : bool =
  printf "%s (y/n) " question
  let key = Console.ReadKey()
  printfn ""
  key.Key = ConsoleKey.Y
